pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -173,6 +201,26 @@ func (l *ListOptionIndexer) afterUpsert(key string, obj any, tx db.TXClient) err
 	if err != nil {
 		return &db.QueryError{QueryString: l.addFieldQuery, Err: err}
 	}
+	return l.upsertLabels(tx, key, obj)
+}
+
+func (l *ListOptionIndexer) upsertLabels(tx db.TXClient, key string, obj any) error {
+	return l.UpsertLabels(tx, l.upsertLabelsStmt, key, obj, l.GetShouldEncrypt())
+}
+
+func (l *ListOptionIndexer) UpsertLabels(tx db.TXClient, stmt *sql.Stmt, key string, obj any, shouldEncrypt bool) error {
```

Comment:
Two observations:

1. it seems like upsertLabels is only really called here, and does not add much to UpsertLabels. I'd suggest to unify the two methods into one
2. I would suggest to uniform how afterUpsert is structured, by having it call two methods: one that encapsulates current functionality (basically `INSERT INTO "%s_fields"`) and another one for the new bits (`upsertLabels`)

That way it should be easier to read and possible to test in isolation as well.


Same for `afterDelete` below.

Reply:
Friday work…

Reply:
Yes. And `afterUpsert` and `afterDelete` are bad names for the actual methods.